pkg/csi/service/common/vsphereutil.go

```diff
@@ -240,17 +240,23 @@ func CreateBlockVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsCluste
 		createSpec.Profile = append(createSpec.Profile, profileSpec)
 	}
 
+	// Encrypt newly provisioned volume or perform a shallow recrypt when created from a snapshot.
```

Comment:
Do we need to put this code block under the check `if spec.ContentSourceSnapshotID != "" {` below? as this is applicable to only for the new volume getting provisioned from snapshot.

Reply:
ignore this comment. I see a check below at line 253.